compute_endpoint/tests/unit/test_engines.py

```diff
@@ -73,6 +73,19 @@ def test_engine_submit(engine_type: GlobusComputeEngineBase, engine_runner):
     assert future.result(timeout=5) == param * 2
 
 
+@pytest.mark.parametrize(
+    "engine_type", (ProcessPoolEngine, ThreadPoolEngine, GlobusComputeEngine)
+)
+def test_engine_working_dir(engine_type: GlobusComputeEngineBase, engine_runner):
+    """working dir remains constant across multiple fn invocations"""
+    engine = engine_runner(engine_type)
+
+    resource_spec = {}
+    future1 = engine._submit(get_cwd, resource_spec)
+    future2 = engine._submit(get_cwd, resource_spec)
+    assert future1.result() == future2.result()
+
```

Comment:
I'm not clear on what this verifies.  As near as I'm able to tell at the moment this passes on `main` prior to this PR, and for me, the paths are absolute.  Is there a setup missing?

Reply:
I've updated this test now. The test now uses `engine.submit(<payload>)` that gets executed with the `execute_task` wrapper. This wrapper creates a `working_dir` and chdirs in before running the task which meant that if `working_dir` was a relative path, each subsequent function runs one nested directory deeper. This test makes sure that this nesting doesn't happen.